### PR TITLE
fix(clerk-js): Button variants and color alignments

### PR DESCRIPTION
**Pull request by desiprisg**
## Description

* Refactor button primitive to use `variant` and `colorScheme`
* Simplify session action buttons
* Text color alignments

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by anagstef** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @anagstef, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-lsi42yu3z.clerkstage.dev) | Feb 02, 2024 10:31 AM |

File: packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx [hunk 1]
```diff
@@ -43,12 +43,9 @@ export const SingleSessionActions = (props: SessionActionsProps) => {
         iconElementId={descriptors.userButtonPopoverActionButtonIcon.setId('signOut')}
         icon={SignOut}
         label={localizationKeys('userButton.action__signOut')}
-        variant='secondaryDanger'
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx:46]: This button needs to be red on hover.
Screenshot 2024-02-02 at 12 52 33

File: packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx [hunk 2]
```diff
@@ -96,7 +93,6 @@ export const MultiSessionActions = (props: SessionActionsProps) => {
           icon={SignOut}
           label={localizationKeys('userButton.action__signOut')}
           onClick={handleSignOutSessionClicked(session)}
-          variant='secondaryDanger'
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx:99]: Also, this one needs to be red according to the designs:
Screenshot 2024-02-02 at 12 53 22

File: packages/clerk-js/src/ui/customizables/parseVariables.ts [hunk 3]
```diff
@@ -17,10 +17,13 @@ export const createColorScales = (theme: Theme) => {
...
   const textTertiary = toHSLA(variables.colorTextTertiary) || colors.makeTransparent(textSecondary, 0.4);
 
+  const secondaryScale = colorOptionToHslaLightnessScale(variables.colorSecondary, 'secondary');
+
   return removeUndefinedProps({
     ...colorOptionToHslaLightnessScale(variables.colorPrimary, 'primary'),
     ...colorOptionToHslaAlphaScale(variables.colorPrimary, 'primaryAlpha'),
-    ...colorOptionToHslaLightnessScale(variables.colorSecondary, 'secondary'),
+    ...secondaryScale,
+    colorTextSecondary: textSecondary,
```
Comment (anagstef) [inline packages/clerk-js/src/ui/customizables/parseVariables.ts:26]: Shouldn't we also pass `colorTextLabel` here?

**Review by anagstef — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx:99]: No it shouldn't.

Comment (desiprisg) [reply packages/clerk-js/src/ui/components/UserButton/SessionActions.tsx:99]: I'd assume it should be, yes, but I should also do that for the sign out all, it probably got away.

File: packages/clerk-js/src/ui/customizables/parseVariables.ts (hunk 3 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/customizables/parseVariables.ts:26]: Yeah I missed that, thanks

**Comment by desiprisg** [comment]
!preview

**Comment by octoper** [comment]
!preview

**Comment by desiprisg** [comment]
!preview

**Review by anagstef — APPROVED** [review]
💯

**Pull request merged**